benchmarks/gpt_fast/benchmark.py

```diff
@@ -18,57 +21,172 @@ class Experiment:
     actual: float
 
 
-def do_inference(mod, x, num_samples: int = 5):
-    total_time = 0
-    start = -1
+class SimpleMLP(nn.Module):
+    def __init__(self, input_dim, hidden_dim, output_dim, dtype):
+        super().__init__()
+        self.fc1 = nn.Linear(input_dim, hidden_dim, dtype=dtype)
+        self.ln1 = nn.LayerNorm(hidden_dim, dtype=dtype)
+        self.fc2 = nn.Linear(hidden_dim, output_dim, dtype=dtype)
+        self.ln2 = nn.LayerNorm(output_dim, dtype=dtype)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = F.gelu(self.ln1(x))
+        x = self.fc2(x)
+        x = F.gelu(self.ln2(x))
+        return x
+
+
+def run_mlp_layer_norm_gelu():
+    dtype_memory_bandwidth_map = {
+        torch.bfloat16: "1805",
+    }
+    input_shapes = [1024, 4096, 8192, 16384]
+    intermediate_size = 14336
+    results = []
+    for dtype, expected_memory_bandwidth in dtype_memory_bandwidth_map.items():
+        memory_bandwidth = 0
+        for D in input_shapes:
+            mod = SimpleMLP(
+                input_dim=D, hidden_dim=intermediate_size, output_dim=D, dtype=dtype
+            ).to("cuda")
+
+            x = torch.randn(D, device="cuda", dtype=torch.bfloat16)
+
+            compiled_mod = torch.compile(mod)
+
+            for _ in range(WARMUP_ITER):
+                compiled_mod(x)
+
+            us_per_iter = do_bench(lambda: compiled_mod(x)) * 1000
+            memory_bandwidth += (
+                (1e6 / us_per_iter) * 4 * D * intermediate_size * dtype.itemsize / 1e9
```

Comment:
I don't get how this is calculatred. For SimpleMLP I think it's better to report flops achieved or something like that.

Reply:
Do you know which is the standard helper function that we used to get flops like ```triton.do_bench```?

Reply:
Oh, nvm, I think it just need to calculate it by my hand.

Reply:
You can use `torch.utils.flop_counter`